packages/badge/components/badgeButton.tsx

```diff
@@ -13,38 +13,31 @@ export interface BadgeButtonProps {
     | "outline";
   onClick: (event?: React.SyntheticEvent<HTMLElement>) => void;
   /**
-   * Tab index indicates if an element can be focuesed for more information see
+   * Tab index indicates if an element can be focused for more information see
    * https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex
    * browser default value for this is -1
    */
   tabIndex?: number;
   children: JSX.Element | string;
 }
 
-export class BadgeButton extends React.PureComponent<BadgeButtonProps, {}> {
-  public static defaultProps: Partial<BadgeButtonProps> = {
-    appearance: "default",
-    tabIndex: -1
-  };
+const BadgeButton = ({
+  appearance = "default",
+  children,
+  onClick,
+  tabIndex = -1
+}: BadgeButtonProps) => {
+  const className = css`
+    outline: none;
+    cursor: pointer;
+    ${badgeButton(appearance)};
+  `;
 
-  public render() {
-    const { appearance, children, onClick, tabIndex } = this.props;
-    const props = {
-      action: onClick,
-      tabIndex
-    };
-
-    const className = css`
-      outline: none;
-      cursor: pointer;
-      ${badgeButton(appearance)};
-    `;
-    return (
-      <Clickable {...props} data-cy="badgeButton">
-        <span className={className}>{children}</span>
-      </Clickable>
-    );
-  }
-}
+  return (
+    <Clickable action={onClick} tabIndex={tabIndex} data-cy="badgeButton">
+      <span className={className}>{children}</span>
+    </Clickable>
+  );
+};
```

Comment:
same goes here. We'll have to memoize it.

Reply:
my guess is that the memoization is the whole reason these are classes (so he could leverage PureComponent)

Reply:
I was thinking about this. I am curious about everyone's opinion here in regard to the PureComponent or memo. It seems like it may be more of a benefit to have those added if there is complex business logic, the component is doing calculations on every render, or the component is displaying some janky behaviors. I am not sure if this would be the case with something like a Badge component. Although I am not sure on the other hand if the benefits outweigh to optimize anyway by adding the useMemo logic to all the components instead if we want functional components. Thoughts? 🤔